docs/technical_specs/model_libraries/generic/unit_models/skeleton_unit.rst

```diff
@@ -0,0 +1,67 @@
+Skeleton Unit Model
+===============================
+
+The IDAES Skeleton Unit model represents a bare bones unit (hence, the name skeleton) that lets the user define a unit model with a custom set
+of variables and constraints that best suits their needs but that can be connected to other models from the unit model library. This unit can be used
+to either define a surrogate for a complex unit model or to define a first-principles based model that does not use IDAES
+ControlVolumes or property packages. To facilitate connection to other models in the unit model library, the user needs to add ports and populate with variables that match
+the port members of the models this will be connected to. To enable this, a special method is provided to enable adding ports with a specified set of variables.
+
+The two methods available for this unit are:
+
+1. ``add_ports``: Allows the user to define the inlet and outlet ports for the model and populate the port with a set of variables.
+2. ``initialize``: A default initialize method that will attempt to solve the model as defined or that can be substituted with a custom callback function provided by the user.
+
+Degrees of Freedom
+------------------
+
+There are no variables or constraints as this is a bare minimum skeleton that will be populated with variables and constraints
+as defined by the user. Therefore, the number of degrees of freedom will be determined based on the model the user implements.
+
+Model Structure
+---------------
+
+Given that this unit is only a skeleton, the model structure is whatever the user intends to be. For example, the
+model has no restrictions on number of inlets and outlets.
+
+Variables
+---------
+
+Variables will be defined by the user. Note that when defining variables, they will need to be indexed by the flowsheet time index irrespective if it is steady-state
+or dynamic. This is to ensure that port members can be connected as there is still a 0 time index even when the flowsheet is steady-state.
+
+Custom Callback
+---------------
+
+Note that the default value for the ``initializer`` config argument is set to trigger the ``_default_initialize`` method. To use a custom callback instead, the user needs to first define a function
+and set the ``obj.config.initializer`` to the custom callback function they defined. In the ``initialize`` method, after checking that degrees of freedom is zero, the following is triggered:
+
+.. code-block:: Python
+
+   self.config.initializer(self, opt=opt, init_log=init_log, solve_log=solve_log)
```

Comment:
It might be worth noting the expectation of a static method and the fact that self here will be the skeleton model object. Given the range of users we expect, it might even be worth showing some pseudo-code of how to initialize a variable.

Reply:
There is no need for a user's custom function to be a static method - a "global" (unbound) function works as well.  I agree that we should clarify semantics in the documentation prototype and not use `self` here.

Reply:
Before the code line, I do mention that "In the ``initialize`` method, after checking that degrees of freedom is zero, the following is triggered:". 
@jsiirola I am not sure I follow why self is not appropriate here.

Reply:
@jghouse88: good point; you are right.